lightning-c-bindings/README.md

```diff
@@ -0,0 +1,82 @@
+   flag is set.
+
+   You MUST NOT create such wrapper structs manually, relying instead on constructors which have
+   been mapped from equivalent Rust constructors.
+
+ * Traits are mapped into a concrete struct containing a void pointer for your use and a jump table
+   listing the functions which the trait must implement. The void pointer may be set to any value
+   and is never interpreted (or dereferenced) by the bindings logic in any way. You may wish to use
+   it as a pointer to your own internal datastructure, though it may also occasionally make sense
+   to e.g. case a file descriptor into a void pointer and use it to track a socket. It is passed as
+   the first argument to all function calls in the trait.
+
+   Rust native structs which implement a trait result in the generation of an X_as_Y function which
+   allows you to use the native Rust object in place of the trait. Such generated objects are only
+   valid as long as the original Rust native object is owned by a C-wrapped struct, and has not been
+   free'd or moved as a part of a Rust function call.
+
+ * Rust "unitary" enums are mapped simply as an equivalent C enum, however some Rust enums have
+   variants which contain payloads. Such enums are mapped automatically by cbindgen as a tag which
+   indicates the type and a union which holds the relevant fields for a given tag. A X_free
+   function is provided for the enum as a whole which automatically frees the correct fields for a
+   given tag, and a Sentinel tag is provided which causes the free function to do nothing (but
+   which must never appear in an enum when accessed by Rust code). The Sentinel tag is used by the
+   C++ wrapper classes to allow moving the ownership of an enum while invalidating the old copy.
+
+ * Struct member functions are mapped as Struct_function_name and take a reference to the mapped
+   struct as their first argument. Free-standing functions are mapped simply as function_name and
+   take the relevant mapped type arguments.
+
+   Functions may return a reference to an underlying Rust object with a mapped struct or may return
+   an owned Rust object with the same. The corresponding free function will Do The Right Thing
+   based on the reference flag contained in the object, but you should reference the Rust
+   documentation for the official reference on the type returned.
+
+As the bindings are auto-generated, the best resource for documentation on them is the native Rust
+docs available via `cargo doc` or [docs.rs/lightning](https://docs.rs/lightning).
+
+The memory model is largely the Rust memory model and not a native C-like memory model. Thus,
+function parameters are largely only ever passed by reference or by move, with pass-by-copy
+semantics only applying to primitive types. However, because the underlying types are largely
+pointers, the same function signature may imply two different memory ownership semantics. Thus, you
```

Comment:
I'm not sure what is meant by the "same function signature" here. This could definitely benefit from some sample code as per my earlier comment.